src/main/resources/uischema/createDirectory.json

```diff
@@ -43,6 +43,28 @@
                                         "required": "true",
                                         "helpTip": "Path to new directory."
                                     }
+                                },
+                                {
+                                    "type": "attribute",
+                                    "value": {
+                                        "name": "autoCreate",
+                                        "displayName": "Auto Create Parent Directories",
+                                        "inputType": "booleanOrExpression",
+                                        "defaultValue": "false",
+                                        "required": "false",
+                                        "helpTip": "Automatically create missing parent directories if they don't exist."
+                                    }
+                                },
+                                {
+                                    "type": "attribute",
+                                    "value": {
+                                        "name": "permissions",
+                                        "displayName": "Permissions",
+                                        "inputType": "stringOrExpression",
+                                        "defaultValue": "",
+                                        "required": "false",
+                                        "helpTip": "Permissions to set on the new directory in octal format (e.g., 755, 644). Mainly supported for local file systems."
+                                    }
                                 }
```

Comment:
shall we use a dropdown with user friendly values something like the following ?

| Drop Down Value                                   | Octal Value |
|---------------------------------------------|------------|
| Read and Write (Only This User)                       | 700        |
| Read and Write (This User), Read Only (Others)       | 755        |
| Read and Write (This User), Read Only (Group)        | 750        |